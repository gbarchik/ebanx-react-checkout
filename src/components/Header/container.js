import { connect } from 'react-redux'

import Header from './index'

const mapStateToProps = (state) => {
    const { config, checkout } = state

    return ({
        config,
        checkout
    })
}

export default connect(mapStateToProps)(Header)