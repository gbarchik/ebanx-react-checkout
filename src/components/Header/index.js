import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import GridContainer from '../GridContainer'

import {
    globalPadding,
    globalMargin,
    headerFontFamily,
    headerFontSizes,
    headerFontWeight
} from '../../helpers/foundation'
import { calculateRem, calculateEm } from '../../helpers/helper'

const StyledHeader = styled.header`
    padding: ${calculateRem(4 * globalPadding)} 0 ${calculateRem(globalPadding)}
`

StyledHeader.displayName = 'Header'

const HeaderTitle = styled.h1`
    margin: 0 0 ${calculateEm((1.5 * globalMargin), headerFontSizes.h1)};
    font-family: ${headerFontFamily};
    font-size: ${calculateRem(headerFontSizes.h1)};
    font-weight: ${headerFontWeight};
`

HeaderTitle.displayName = 'HeaderTitle'

const Header = (props) => {
    const { title, description } = props.config
    const { bought } = props.checkout

    return (
        <StyledHeader>
            <GridContainer>
                <HeaderTitle>{title}</HeaderTitle>
                {description && !bought && <p>{description}</p>}
            </GridContainer>
        </StyledHeader>
    )
}
    

Header.propType = {
    config: PropTypes.shape({
        title: PropTypes.string.isRequired,
        description: PropTypes.string
    }),
    checkout: PropTypes.shape({
        bought: PropTypes.bool.isRequired
    })
}

export default Header
