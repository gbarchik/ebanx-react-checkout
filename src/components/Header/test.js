import React from 'react'
import { shallow } from 'enzyme'

import Header from './index'

const fakeProps = {
    config: {
        title: 'E-book Store',
        description: 'Welcome to the best place'
    },
    checkout: {
        bought: false
    }
}

describe('(Layout) Header', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Header {...fakeProps} />)
        wrapper.setProps({
            config: {
                title: ''
            }
        })
        expect(wrapper).toMatchSnapshot()
    })

    it('renders as a <header>', () => {
        const wrapper = shallow(<Header {...fakeProps} />)
        expect(wrapper.html()).toMatch(/^<header/)
    })

    it('should render a <HeaderTitle> title', () => {
        const wrapper = shallow(<Header {...fakeProps} />)
        expect(wrapper.find('HeaderTitle')).toHaveLength(1)
    })

    it('should render the Title', () => {
        const wrapper = shallow(<Header {...fakeProps} />)
        wrapper.setProps({ config: { title: 'EBANX' } })
        expect(wrapper.find('HeaderTitle').children().text()).toBe('EBANX')
    })

    it('should show the description if it is not the ThankYou page', () => {
        const wrapper = shallow(<Header {...fakeProps} />)
        expect(wrapper.find('p')).toHaveLength(1)
    })

    it('should not show the description if it is the ThankYou page', () => {
        const wrapper = shallow(<Header {...fakeProps} />)
        wrapper.setProps({ checkout: { bought: true } })
        expect(wrapper.find('p')).toHaveLength(0)
    })
})
