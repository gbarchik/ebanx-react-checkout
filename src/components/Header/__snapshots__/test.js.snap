// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Layout) Header renders correctly 1`] = `
<Header>
  <styled.div>
    <HeaderTitle />
  </styled.div>
</Header>
`;
